src/System.Linq/tests/UnionTests.cs

```diff
@@ -413,5 +414,34 @@ public void HashSetWithBuiltInComparer_HashSetContainsNotUsed()
             Assert.Equal(new[] { "A", "a" }, input2.Union(input1, EqualityComparer<string>.Default));
             Assert.Equal(new[] { "A" }, input2.Union(input1, StringComparer.OrdinalIgnoreCase));
         }
+
+
+        private class MyEnumerable : IEnumerable<long>
+        {
+            private MyEnumerator _enumerator;
+            public bool DisposeCalledOnEnumerator => _enumerator.DisposeCalled;
+            public IEnumerator<long> GetEnumerator() => (_enumerator =  new MyEnumerator());
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        }
+
+        private class MyEnumerator : IEnumerator<long>
+        {
+            public bool DisposeCalled { get; private set; }
+            public long Current => 0;
+            object IEnumerator.Current => Current;
+            public bool MoveNext() => false;
+            public void Reset() { }
+            public void Dispose() { DisposeCalled = true; }
+        }
```

Comment:
Good point, will change!
Is there any situation where it would require the Enumerable to be Generic? Wouldn't it be enough to just inherit from IEnumerable<(some_random_type)>?